pkg/cmd/roachtest/testselector/snowflake_query.sql

```diff
@@ -36,14 +40,14 @@ select
   case when
          -- mark as selected if
          failure_count > 0 or -- the test has failed at least once in the past "forPastDays" days
-         first_run > dateadd(DAY, ?, current_date()) or -- recently added test - test has not run for more than "firstRunOn" days
-         last_run < dateadd(DAY, ?, current_date()) or -- the test has not been run for last "lastRunOn" days
-         last_run is null -- the test is always ignored till now or have never been run
+         first_run > dateadd(DAY, ?, ts.t) or -- recently added test - test has not run for more than "firstRunOn" days
+         last_run < dateadd(DAY, ?, ts.t) or -- the test has not been run for last "lastRunOn" days
+         last_run is null or -- the test is always ignored till now or have never been run
+         (last_status='UNKNOWN' and recent_ignore_details!='test selector') -- the test has not been ignored by test selector
```

Comment:
We want to select all tests that are not ignored by test selector.

UNKNOWN can be for the following scenarios -
test is run, but added to skip by the test writer (the condition check above) - In this scenario, we still want to keep selecting the test as we do not know when user may mark it to not skipped
test is never run due to incompatibility - here also, user can make it compatible at any point. So, we always select.
test is never run - The status will be UNNOWN for all entries. So, just checking the last entry may not help ans we want to check the last_run is null
test is not run by test selector in the last run - This can should not be by default selected. Let the selector decide based on the last_run sorted data.

If I do not check for UNKNOWN, all tests with last_status SUCCESS will also be by default selected as recent_ignore_details is not "test selector" for SUCCESS ones. This is not right.

Reply:
Actually, you are right in the last_run condition - we can remove that condition as the last_status='UNKNOWN' and recent_ignore_details!='test selector') condition shadows that.

Reply:
I have added the above as a comment in the code for more clarity and also removed the last_run is null condition

Reply:
Okay! Thanks for explaining!